From 92a1e558d984ff61b144dbb6a7dd4b4427406921 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Tue, 19 Apr 2011 15:27:39 +0800
Subject: [PATCH 141/256] ENGR00142312 ipulib: export the IPC initial function
 to android fb HAL

CTS test forbid the write permission for other group to shared memory
directory: /mnt/shm, so we must create the IPU semaphore share files
on boot when framebuffer mapped by the system server who's id is system:system.

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 ipu/mxc_ipu_hl_lib.c | 8 ++++----
 ipu/mxc_ipu_hl_lib.h | 7 +++++++
 2 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/ipu/mxc_ipu_hl_lib.c b/ipu/mxc_ipu_hl_lib.c
index da063a6..ab70e4c 100644
--- a/ipu/mxc_ipu_hl_lib.c
+++ b/ipu/mxc_ipu_hl_lib.c
@@ -2424,7 +2424,7 @@ static void * _get_shm(char *name, int size, int *first)
 	return buf;
 }
 
-static int _ipu_ipc_prepare(void)
+int mxc_ipu_lib_ipc_init(void)
 {
 	int ret = 0;
 	int first = 0;
@@ -2631,7 +2631,7 @@ int mxc_ipu_lib_task_init(ipu_lib_input_param_t * input,
 	}
 
 	if (!g_ipu_shm) {
-		if (_ipu_ipc_prepare() < 0) {
+		if (mxc_ipu_lib_ipc_init() < 0) {
 			ret = -1;
 			goto err0;
 		}
@@ -3414,7 +3414,7 @@ int mxc_ipu_lib_task_control(int ctl_cmd, void * arg, ipu_lib_handle_t * ipu_han
 		ipu_lib_ctl_task_t * ctl_task =
 				(ipu_lib_ctl_task_t *) arg;
 		if (!g_ipu_shm) {
-			if (_ipu_ipc_prepare() < 0) {
+			if (mxc_ipu_lib_ipc_init() < 0) {
 				ret = -1;
 				goto done;
 			}
@@ -3433,7 +3433,7 @@ int mxc_ipu_lib_task_control(int ctl_cmd, void * arg, ipu_lib_handle_t * ipu_han
 		ipu_lib_priv_handle_t * ipu_priv_handle;
 
 		if (!g_ipu_shm) {
-			if (_ipu_ipc_prepare() < 0) {
+			if (mxc_ipu_lib_ipc_init() < 0) {
 				ret = -1;
 				goto done;
 			}
diff --git a/ipu/mxc_ipu_hl_lib.h b/ipu/mxc_ipu_hl_lib.h
index b643193..547f845 100644
--- a/ipu/mxc_ipu_hl_lib.h
+++ b/ipu/mxc_ipu_hl_lib.h
@@ -411,6 +411,13 @@ int mxc_ipu_lib_task_buf_update(ipu_lib_handle_t * ipu_handle,
  */
 int mxc_ipu_lib_task_control(int ctl_cmd, void * arg, ipu_lib_handle_t * ipu_handle);
 
+/*!
+ * This function initialize the ipu IPC share memory.
+ *
+ * @return 0 as successful, -1 failed
+ */
+int mxc_ipu_lib_ipc_init(void);
+
 #ifdef __cplusplus
 }
 #endif
-- 
1.8.0

